sklearn/neighbors/_binary_tree.pxi

```diff
@@ -979,6 +978,19 @@ cdef class BinaryTree:
             self.node_bounds.base,
         )
 
+    @property
+    def valid_metric(self):
```

Comment:
`valid_metric` must be renamed to `valid_metrics`.

Also I think `valid_metrics` better be a [`classmethod`](https://docs.python.org/3/library/functions.html#classmethod):
```suggestion
    @classmethod
    def valid_metrics(cls):
```

This way, it can be called without creating instance, e.g. this works:

```python
BallTree.valid_metrics()
```

Reply:
The problem with using `valid_metrics()` instead of `valid_metric()` is that it conflicts with the variable `valid_metrics` inside the class, and thus throws an error while importing the class `KDTree`.

Reply:
You can prefix the variable inside the class with `_`.

Reply:
Doing that, we would have to change the variable name at each location where `valid_metrics` is accessed, to `_valid_metrics`.

Reply:
I think that should be fine @01-vyom. Perhaps we can try it out and see if any errors are raised upon building docs and unit tests?